### PR TITLE
Technical/nijo 1322 add agency id to agent markup2

### PR DESCRIPTION
**Pull request by Raibo**
This is second part of a partial pull request.
Added `agencyId` into all read/write places of `MarkupPolicies` table when dealing with agents markups.
Also added audit logging to admin service and refactored scope validation a bit.

### CODE REVIEW

File: Api/AdministratorServices/AgentService.cs [hunk 1]
```diff
@@ -37,8 +37,8 @@ public async Task<Result<List<SlimAgentInfo>>> GetAgents(int agencyId)
                        Name = name,
                        Created = created,
                        IsActive = relation.IsActive,
-                       MarkupSettings = (agent.DisplayedMarkupFormula != null)
-                           ? agent.DisplayedMarkupFormula
+                       MarkupSettings = !string.IsNullOrWhiteSpace(relation.DisplayedMarkupFormula)
```
Comment (mingazhev) [inline Api/AdministratorServices/AgentService.cs:40]: Have you tested that this works?

File: Api/Controllers/AgentControllers/AgentMarkupsController.cs [hunk 2]
```diff
@@ -111,7 +111,8 @@ public async Task<IActionResult> ModifyPolicy(int agentId, int policyId, [FromBo
         [ProducesResponseType(typeof(List<MarkupInfo>), (int) HttpStatusCode.NoContent)]
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentMarkupsController.cs:111]: NoContent looks invalid

File: HappyTravel.Edo.Data/Migrations/20210219145917_MoveMarkupFormulaToRelation.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace HappyTravel.Edo.Data.Migrations
+{
+    public partial class MoveMarkupFormulaToRelation : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropColumn(
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Migrations/20210219145917_MoveMarkupFormulaToRelation.cs:9]: I think you should add a script that copies existing values. Or at least look at the production to be sure that we do not lose any data

File: Api/AdministratorServices/AgentService.cs (hunk 1 above)
Comment (Raibo) [reply Api/AdministratorServices/AgentService.cs:40]: `string.IsNullOrWhiteSpace`?
Numerous times.
This particular place in code - no, but `!string.IsNullOrWhiteSpace(s)` generally replaces `s != null` well.

Comment (Raibo) [reply Api/AdministratorServices/AgentService.cs:40]: Takled a bit in discord - question was about convertion of this to SQL.
Answer - `!string.IsNullOrWhiteSpace` indeed converts to SQL and works fine.

File: HappyTravel.Edo.Data/Migrations/20210219145917_MoveMarkupFormulaToRelation.cs (hunk 3 above)
Comment (Raibo) [reply HappyTravel.Edo.Data/Migrations/20210219145917_MoveMarkupFormulaToRelation.cs]: Done

File: Api/Controllers/AgentControllers/AgentMarkupsController.cs (hunk 2 above)
Comment (Raibo) [reply Api/Controllers/AgentControllers/AgentMarkupsController.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**